UnityProject/Assets/Scripts/Messages/Client/Admin/RequestUpdatePlayerCharacterSheet.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using Logs;
+using Messages.Server;
+using Mirror;
+using Newtonsoft.Json;
+using Systems.Character;
+using UI.Admin.DIMGUI.Characters;
+using UnityEngine;
+
+namespace Messages.Client.Admin
+{
+	public class RequestUpdatePlayerCharacterSheet : ClientMessage<RequestUpdatePlayerCharacterSheet.NetMessage>
+	{
+		public struct NetMessage : NetworkMessage
+		{
+			public string CharacterSheetJson;
+			public string AccountID;
+		}
+
+		public override void Process(NetMessage msg)
+		{
+			PlayerInfo account = null;
+			foreach (var player in PlayerList.Instance.AllPlayers)
+			{
+				if (player.AccountId == msg.AccountID)
+				{
+					account = player;
+					break;
+				}
+			}
+
+			if (account == null)
+			{
+				Loggy.LogError("[RequestUpdatePlayerCharacterSheet] Could not find account to update, or lacking admin permissions.");
+				UpdateTheRequestToCharacterSheetUpdateToRequests.SendSheetUpdate(SentByPlayer, false);
+				return;
+			}
+
+			try
+			{
+				CharacterSheet sheet = JsonConvert.DeserializeObject<CharacterSheet>(msg.CharacterSheetJson);
+				account.Mind.CurrentCharacterSettings = sheet;
+				UpdateTheRequestToCharacterSheetUpdateToRequests.SendSheetUpdate(SentByPlayer);
+			}
+			catch (Exception e)
+			{
+				if (account.Mind == null)
+				{
+					Loggy.LogError("[RequestUpdatePlayerCharacterSheet] Could not find a mind to link sheet to.");
+				}
+				else
+				{
+					Loggy.LogError(e.ToString());
+				}
+				UpdateTheRequestToCharacterSheetUpdateToRequests.SendSheetUpdate(SentByPlayer, false);
+			}
+		}
+
+		public static NetMessage SendSheetUpdate(string userIDToUpdate, string characterSheetJson)
+		{
+			var msg = new NetMessage
+			{
+				CharacterSheetJson = characterSheetJson,
+				AccountID = userIDToUpdate,
+			};
+
+			Send(msg);
+			return msg;
+		}
+	}
+
+	public class UpdateTheRequestToCharacterSheetUpdateToRequests : ServerMessage<UpdateTheRequestToCharacterSheetUpdateToRequests.NetMessage>
+	{
+		public struct NetMessage : NetworkMessage
+		{
+			public bool IsSuccess;
+		}
+
+		public override void Process(NetMessage msg)
+		{
+			//(Max): This sucks massive balls for performance. but we can't use netid on UI.
+			// at least the client is the one that will handle this, not the server.
+			var obj = GameObject.FindObjectsByType<CharacterSheetEditor>(FindObjectsSortMode.InstanceID);
```

Comment:
Singleton?

Reply:
why a Singleton for a single non-manager object that is probably going to be destroyed at some point?

Reply:
ois it going to be destroyed

Reply:
your code suggests you expect it to be there. If you were getting or creating the object, I'd understand but as is, it seems like a singleton would be better because it seems this is in the scene hierarchy to begin with, is it not?

Where are you adding the gameobject that has the component?

Reply:
> Where are you adding the gameobject that has the component?

It's on admin tooling. (the window we use for managing players and what not)

Reply:
maybe make the admin stuff universally accesible as a singleton and expose a reference to this window. You will likely need this very same pattern for many other things you are going to add there.

Reply:
Using an instance now